app/grandchallenge/evaluation/views/api.py

```diff
@@ -23,3 +46,118 @@ class EvaluationViewSet(ReadOnlyModelViewSet):
         *api_settings.DEFAULT_RENDERER_CLASSES,
         PaginatedCSVRenderer,
     )
+
+    @action(
+        detail=True, methods=["PATCH"], permission_classes=[CanClaimEvaluation]
+    )
+    def claim(self, request, *args, **kwargs):
+        evaluation = self.get_object()
+
+        if not evaluation.status == Evaluation.PENDING:
+            return Response(
+                {"status": "You can only claim pending evaluations."},
+                status=400,
+            )
+
+        if request.user.claimed_evaluations.exists():
+            return Response(
+                {"status": "You can only claim one evaluation at a time."},
+                status=400,
+            )
+
+        evaluation.status = Evaluation.CLAIMED
+        evaluation.claimed_by = request.user
+        evaluation.started_at = now()
+        evaluation.save()
+
+        serializer = ExternalEvaluationSerializer(
+            instance=evaluation, context={"request": request}
+        )
+        return Response(serializer.data)
+
+    @action(detail=False, methods=["GET"])
+    def claimable_evaluations(self, request, *args, **kwargs):
+        queryset = get_objects_for_user(
```

Comment:
You can probably reuse `permission_classes=[CanClaimEvaluation]`

Reply:
No, I would need to write a custom filter, but since we only need this here, I think doing it inside the view is clearer.

Reply:
You can use then permission class then filter afterwards, right?

Reply:
Setting the permission_class doesn't filter the queryset by permission. Normally on a list view, the `ObjectPermissionsFilter` does that, so I'd need a custom version of that filter to achieve this, but that's not more efficient than what I'm doing now, right?